src/components/Transfer.tsx

```diff
@@ -62,44 +61,53 @@ function Transfer() {
     transferAmountError,
     receiverAddress,
     estimatedFeeLoading,
-    transactionRunning
+    transactionRunning,
+    transactionReadyToExecute
   } = useTransactionContext();
+
   const { api } = sourceChainDetails.apiConnection;
   const balance = useBalance(api, account?.address || '');
 
-  const { isButtonDisabled, sendLaneMessage } = useSendMessage({
+  const dispatchCallback = useCallback(
+    (value: string) => {
+      dispatchTransaction(
+        TransactionActionCreators.setTransferAmount(
+          value !== null ? value?.toString() : '',
+          api.registry.chainDecimals[0]
+        )
+      );
+    },
+    [api.registry.chainDecimals, dispatchTransaction]
+  );
+
+  const [currentInput, setInput] = useDebounceState({ initialValue: '0', dispatchCallback });
+
+  const { sendLaneMessage } = useSendMessage({
     input: transferAmount?.toString() ?? '',
     type: TransactionTypes.TRANSFER
   });
 
   const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const actualValue = event.target.value;
-    setInput(actualValue);
-    dispatchTransaction(
-      TransactionActionCreators.setTransferAmount(
-        actualValue !== null ? actualValue?.toString() : '',
-        api.registry.chainDecimals[0]
-      )
-    );
+    setInput(event.target.value);
   };
 
   useEffect((): void => {
     transactionRunning && setInput('');
-  }, [transactionRunning]);
+  }, [setInput, transactionRunning]);
 
   useEffect((): void => {
     estimatedFee &&
       transferAmount &&
-      setAmountNotCorrect(new BN(balance.free).sub(transferAmount).add(new BN(estimatedFee)).isNeg());
-  }, [transferAmount, estimatedFee, api.registry.chainDecimals, balance.free]);
+      setAmountNotCorrect(new BN(balance.free).sub(new BN(transferAmount).add(new BN(estimatedFee))).toNumber() < 0);
```

Comment:
This should not have been changed it basically reverts all the hard work Nikos put into this in #204

Reply:
ouh! that came bacause a conflict resolution that got the old code.